fsw/src/crypto_props.c

```diff
@@ -0,0 +1,351 @@
+            }
+            
+     
+        }
+        //head is NOT NULL 
+        else
+        {
+            crypto_config_node* p_current = (*pp_self)->p_head; 
+            crypto_config_node* p_previous = NULL; 
+            while (NULL!=p_current)
+            {
+                p_previous = p_current; 
+                p_current = p_current->p_next; 
+                
+            }//end while 
+            
+            //add a new node to the end 
+            p_previous->p_next = calloc(1,sizeof(crypto_config_node));
+            p_previous->p_next->key = calloc((*pp_self)->KEY_SIZE_IN_BYTES,sizeof(char)); 
+            p_previous->p_next->value = calloc((*pp_self)->VALUE_SIZE_IN_BYTES,sizeof(char));
+            //set key & value
+            if (safe_copy(p_previous->p_next->key,key) && safe_copy(p_previous->p_next->value,value))
+            {
+                strcpy(p_previous->p_next->key,key);
+                strcpy(p_previous->p_next->value,value);
+                success = 1;
+            }
+            else
+            {
+                printf("ERROR crypto_config,crypto_"
+                        "config_props_add_property_helper(), prevented unsafe "
+                        "copy.\n");
+                printf("key size=%lu, key destination size=%lu,value size=%lu, "
+                        "value destination size=%lu.\n",sizeof(key),
+                        sizeof((*pp_self)->p_head->key),
+                        sizeof(value),
+                        sizeof((*pp_self)->p_head->value)); 
+            }
+           
+        }
+    }
+    else
+    {
+        printf("ERROR, crypto_config_props_add_property_helper() inputs are empty or null.\n");
+    }
+    
+    return success; 
+}
+
+int crypto_config_set_property_value(char* key, char* value)
+{
+    return crypto_config_props_set_property_value_helper(key,value,&p_self);
+}
+
+int crypto_config_props_set_property_value_helper(char* key, char* value, crypto_config_list** pp_self)
+{
+    int success = 0; 
+    //iterate through the link list until you reach the right node
+    if (NULL!=key && strlen(key)> 0 && NULL!=value && NULL!=pp_self && NULL!=(*pp_self))
+    {
+        crypto_config_node* p_current = (*pp_self)->p_head; 
+        while (p_current!=NULL)
+        {
+            //returns 0 if NOT equal 
+            if (strcmp(p_current->key,key)!=0)
+            {
+                p_current= p_current->p_next; 
+            }
+            else
+            {
+                if (safe_copy(p_current->value,value))
+                {   
+                    //reset old values to empty 
+                    memset(p_current->value, 0,sizeof((*pp_self)->VALUE_SIZE_IN_BYTES));
+                    //replace old value with the new 
+                    strcpy(p_current->value,value);  
+                    success = 1; 
+                }
```

Comment:
On a similar note to my comment above, let's just free the current 'value', dynamically malloc a new char* for the size of the new value, and have that be what's pointed to.

Reply:
Also, why is success=1 here? Seems like it worked?

Reply:
`success= 1` or in the update code` status = true`, because yes it worked.